RecoLocalTracker/SiStripClusterizer/plugins/SiStripClusterToDigiProducer.cc

```diff
@@ -31,19 +31,21 @@ class SiStripClusterToDigiProducer : public edm::stream::EDProducer<> {
 
 private:
   void process(const ClusterCollection& input, std::vector<DetDigiCollection>& output_base);
-  void initialize(const edm::EventSetup& es);
   void setDetId(const uint32_t id);
   float gain(const uint16_t& strip) const { return gainHandle->getStripGain(strip, gainRange); }
   uint16_t applyGain(const uint16_t& strip, const uint16_t& adc);
 
   edm::EDGetTokenT<ClusterCollection> token;
   SiStripApvGain::Range gainRange;
-  edm::ESHandle<SiStripGain> gainHandle;
-  uint32_t gain_cache_id, detId;
+  edm::ESGetToken<SiStripGain, SiStripGainRcd> gainToken_;
+  edm::ESWatcher<SiStripGainRcd> gainWatcher_;
+  const SiStripGain* gainHandle;
```

Comment:
This not being a "handle" any more, it could be probably better renamed without "Handle" in it